### ISSUE TITLE
Hello Lucy not detecting

### ISSUE THREAD

**Issue by varadwattamwar** [mykle1/Hello-Lucy#33]
I completed all the steps successfully but after speaking Hello Lucy there is no response from MM. Mic and speaker is set properly.

**Comment by sdetweil** [comment]
have you tested the mic this way 
```
arecord test.wav
speak something
ctrl-c to stop recording
aplay test.wav
```
do you hear what you said?

**Comment by varadwattamwar** [comment]
> aplay test.wav

Yes, I heard what i said, but there is some noise disturbance also.

**Comment by sdetweil** [comment]
then it will be hard for computer to sound match.. u need to get rid of the noise

**Comment by varadwattamwar** [comment]
> then it will be hard for computer to sound match.. u need to get rid of the noise

Ok alright, need to try with a new mic then.
Thanks for the reply.

**Comment by varadwattamwar** [comment]
> then it will be hard for computer to sound match.. u need to get rid of the noise



> > then it will be hard for computer to sound match.. u need to get rid of the noise
> 
> Ok alright, need to try with a new mic then.
> Thanks for the reply.

Still the result is same , no response from lucy.
One thing I observed is, I also get an error for Hello-Lucy while running MM.
Error:  Failed to fetch git data for Hello-Lucy: Error: block timeout reached
Is this the reason for no response?

**Comment by sdetweil** [comment]
>Error: Failed to fetch git data for Hello-Lucy: Error: block timeout reached
Is this the reason for no response?

the timeout on the updatenotifications module is too short.. fixed in the upcoming release, and in the magicmirror develop branch now

when MagicMirror is running, in a terminal window or ssh session window do 
```
ps -ef | grep arecord | grep -v grep 
```
it should put out a line like this 
```
????     11505 17306  0 05:11 pts/9    00:00:45 arecord -D default -q -r 16000 -c 1 -t wav 
```
if it does not, then it means arecord failed to start 
in my case it is using the default device...
arecord -l  (lower case L, for list) 
will show the devices...  i don't know how Lucy is configured for mic..

**Comment by varadwattamwar** [comment]
> > Error: Failed to fetch git data for Hello-Lucy: Error: block timeout reached
> > Is this the reason for no response?
> 
> the timeout on the updatenotifications module is too short.. fixed in the upcoming release, and in the magicmirror develop branch now
> 
> when MagicMirror is running, in a terminal window or ssh session window do
> 
> ```
> ps -ef | grep arecord | grep -v grep 
> ```
> 
> it should put out a line like this
> 
> ```
> ????     11505 17306  0 05:11 pts/9    00:00:45 arecord -D default -q -r 16000 -c 1 -t wav 
> ```
> 
> if it does not, then it means arecord failed to start
> in my case it is using the default device...
> arecord -l (lower case L, for list)
> will show the devices... i don't know how Lucy is configured for mic..

There is only one device mic shown in the list whose card and device number is 1,0.
Still lucy is not detecting users voice so any other solution?
My Rpi version 3B+.

**Comment by varadwattamwar** [comment]
I tried:
In asoundrc file I changed “hw:1,0” to “plughw:1,0”
But the result is same lucy is not responding.
Please can you give another solution @mykle1

**Comment by sdetweil** [comment]
after changing .asoundrc did u reboot and also do the arecord test.wav/aplay test.wav cycle?

**Comment by varadwattamwar** [comment]
arecord test.wav shows a error:
ALSA lib conf.c:1891:(_snd_config_load_with_include) :3:1:Unexpected }
ALSA lib conf.c:3650:(config_file_open) /home/pi/.asoundrc may be old or corrupted: consider to remove or fix it
ALSA lib conf.c:3572:(snd_config_hooks_call) function snd_config_hook_load returned error: Invalid argument
ALSA lib conf.c:4026:(snd_config_update_r) hooks failed, removing configuration
arecord: main:828: audio open error: Invalid argument

So should I change “plughw:1,0” to “hw:1,0” 

And what should I do to get lucy working??
Please help!!! @sdetweil

**Comment by sdetweil** [comment]
i would change it back to how it was..    and reboot..

i do not know.. voice detection is done by an outside module.. pocket_sphinx_continuous
i stopped using Lucy because it didn't recognize my speech  90% of the time. 

you could add 
```
debug:true 
```
to the config:{} section for the Hello-Lucy module in config.js and
then check the window where u start  MM, either npm start or pm2 (using the pm2 logs --lines=100 command)
to see what, if anything, was recognized.. 

if you use npm start, u can redirect the output to a file , with npm start >somefile 
and then u can look at somefile live or after stopping MM..

**Comment by varadwattamwar** [comment]
I identified the issue .
The issue is lucy is not correctly interpreting my voice.
Ex: If I say "sleep" It interprets "eight"
This is the issue 
Is there any solution for this ? 
Is this the fault of the repository or mic?
But my mic is fine.  

> i would change it back to how it was.. and reboot..
> 
> i do not know.. voice detection is done by an outside module.. pocket_sphinx_continuous
> i stopped using Lucy because it didn't recognize my speech 90% of the time.
> 
> you could add
> 
> ```
> debug:true 
> ```
> 
> to the config:{} section for the Hello-Lucy module in config.js and
> then check the window where u start MM, either npm start or pm2 (using the pm2 logs --lines=100 command)
> to see what, if anything, was recognized..
> 
> if you use npm start, u can redirect the output to a file , with npm start >somefile
> and then u can look at somefile live or after stopping MM..

**Comment by sdetweil** [comment]
according to the cmu sphinx doc, you could try to create a new acoustic model

https://stackoverflow.com/questions/33453691/cmu-sphinx-for-indian-english

which leads to this
https://cmusphinx.github.io/wiki/tutorialadapt/
and here 
https://sourceforge.net/projects/cmusphinx/files/Acoustic%20and%20Language%20Models/

lucy module expects the ldc/lm files to be named Hello-Lucy.dlc and Hello-Lucy.lm
but I think it regenerates based on the word.json file for usage specific  recognition
using lmtool (http://www.speech.cs.cmu.edu/tools/lmtool-new.html)

I don't know how to do the adaptation

**Comment by varadwattamwar** [comment]
> word

Thankyou will try this.

**Comment by cq0q** [comment]
> > word
> 
> Thankyou will try this.

Hi sir, i have the same problem now .. which is when i open my magic mirror it shows up: command something then it shows error .. and i can't say anything ..
Did you solve the problem ? If you did please tell me how

**Comment by TwitchPlaysSpeedruns** [comment]
I have an issue where the screen displays a microphone symbol and the word Error. Lucy speaks on start up but does not respond. I have done the test and can record and playback in cmd. Using Rpi3 and raspian full. 

card 1: Seri [Plantronics Blackwire 3220 Seri], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

I have set the config to both (1,0) and (0,0) and both have the same error. 
"ps -ef | grep arecord | grep -v grep"  does not return anything

Any ideas?

**Comment by roccamorice** [comment]
I have the exact same issue/behaviour as TwitchPlaysSpeedruns. Was this ever solved?

**Comment by sdetweil** [comment]
@roccamorice  did you try arecord test.wav

then aplay test.wav

**Comment by roccamorice** [comment]
Yes. It worked fine.Sent from my iPhoneOn Feb 16, 2024, at 9:26 PM, sam detweiler ***@***.***> wrote:﻿
@roccamorice  did you try arecord test.wav
then aplay test.wav

**Comment by roccamorice** [comment]
Also when I added the Debug: true to the module config, I don't get a debug log although the display that normally shows icon of microphone and ERROR also displays UNDEFINED below the microphone icon.  I only see the error.log in the Hello-Lucy modules directory and the error I keep getting is " Error: spawn pocketsphinx_continuous ENOENT -2".

**Comment by sdetweil** [comment]
@roccamorice  did you do step 3 of the installation

![Screenshot_20240218_092040_Chrome](https://github.com/mykle1/Hello-Lucy/assets/1498142/f3ba57f5-6a14-4f65-b024-e3bafa1b969b)

**Comment by roccamorice** [comment]
Yes I did step 3 and the dependencies.sh executed successfully.

**Comment by sdetweil** [comment]
step 3 should have installed the pocket sphinx continuous command line app.  

it may not have updated the path.  I would see if you can find the executable...

**Comment by roccamorice** [comment]
there is a pocketsphinx-continuous directory in the node_modules directory in Hello-Lucy modules directory.  In this directory there are index.js and package.json files.  I don't think you are referring to these as the executable...or are you?

**Comment by sdetweil** [comment]
well, it looks like they have removed that program.. SO.. rework is required on Hello-Lucy

https://cmusphinx.github.io/2022/08/pocketsphinx-continuous/

**Comment by SinCity64** [comment]
Hello everyone

Any new information on this?

I'm trying to get the project running on Raspberry 5 (Bookworm).

Lucy, however, doesn't want to listen to me ;-). After starting MagicMirror, a microphone icon flashes and next to it is the text "Command Please":

PS: I've completed the basic installation (all 3 steps) and the configuration in piAudioSetup.md and Config.sys. 
The microphone and audio outputs work.

Greetings René

**Comment by sdetweil** [comment]
@SinCity64 did you see this 
https://github.com/mykle1/Hello-Lucy/issues/33#issuecomment-1952813205

no work has been done

**Comment by SinCity64** [comment]
...yes, I saw that.

I was hoping someone had made some progress here.
After x hours of research on the internet, I've learned a lot, but implementing the necessary adjustments is beyond my programming skills.
